File: 452.java
// LeetCode 452 Minimum Number of Arrows to Burst Balloons
// Reference:https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/solutions/93703/share-my-explained-greedy-solution/?envType=study-plan-v2&envId=top-interview-150

import java.util.Arrays;

class Solution {
    public int findMinArrowShots(int[][] points) {

        if (points.length == 0){
            return 0;
        }
        Arrays.sort(points, (a, b) -> a[1] - b[1]);
        int arrowPos = points[0][1];
        int arrowCnt = 1;
        for (int i = 1; i < points.length; i++){
            if (arrowPos >= points[i][0]){
                continue;
            }
            arrowCnt++;
            arrowPos = points[i][1];
        }
        return arrowCnt;
        
    }
}

File: 452.js
// LeetCode 452 Minimum Number of Arrows to Burst Balloons
// Reference:https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/solutions/4890366/beat-93-80-full-explanation-with-pictures/?envType=study-plan-v2&envId=top-interview-150

/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    points.sort((a,b) => a[0] - b[0]);

    let arrows = 1;
    let end = points[0][1];

    for (let i = 1; i < points.length; i++){

        if (points[i][0] > end){
            arrows++;
            end = points[i][1];
        } else {
            end = Math.min(end, points[i][1]);
        }
    }
    return arrows;
};

